cmd/vfkit/main.go

```diff
@@ -264,3 +276,94 @@ func startIgnitionProvisionerServer(ignitionReader io.Reader, ignitionSocketPath
 	log.Debugf("ignition socket: %s", ignitionSocketPath)
 	return srv.Serve(listener)
 }
+
+// it generates a cloud init image by taking the files passed by the user
+// as cloud-init expects files with a specific name (e.g user-data, meta-data) we check the filenames to retrieve the correct info
+// if some file is not passed by the user, an empty file will be copied to the cloud-init ISO to
+// guarantee it to work (user-data and meta-data files are mandatory and both must exists, even if they are empty)
+// if both files are missing it returns an error
+func generateCloudInitImage(files []string) (string, error) {
+	if len(files) == 0 {
+		return "", nil
+	}
+
+	var userDataReader, metaDataReader io.Reader
+	for _, path := range files {
+		if path == "" {
+			continue
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return "", err
+		}
+		defer file.Close()
+
+		// based on the file name we save its content on the corresponding var
+		filename := filepath.Base(path)
+		switch filename {
+		case "user-data":
+			userDataReader = file
+		case "meta-data":
+			metaDataReader = file
+		}
+	}
+
+	if userDataReader == nil && metaDataReader == nil {
+		return "", fmt.Errorf("cloud-init needs user-data and meta-data files to work")
```

Comment:
I'm a bit confused by the error message and the test. If I'm not mistaken, the test implies that we need user-data **or** meta-data to be provided, but the way I understand the error message,  we expect to get both?

Reply:
Cloud-init requires both files to work. They can be empty but they must exist otherwise it won't work.

However I think it's reasonable to allow the user to only pass on file. Maybe you just need to push some customization using the user-data (or the meta-data) and leave the other file empty. This way if we find out that you only have the user-data, we create an empty meta-data file (or vice versa).

On the other hand, if we do not find any of those files, there is most probably an error and the user needs to be informed

Reply:
should the input be a dir that contains files and we check for meta and user data as a minimum and proceed? or is it common to have multiples of those? (when I was a cloud-init expert in the day, it was just mostly those two)

Reply:
i am using this stuff in another tool and I thought the same thing. 
Would it be acceptable to allow the user to pass a folder or files? Or would it be confusing?
`--cloud-init <folder-path>`
and also
`--cloud-init <folder-path>/userdata,<other-folder>/meta-data`